### PR TITLE
Runtime unit tests

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: runtime/src/test_support/cpp/CompilerGenerated.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#ifndef RUNTIME_COMPILER_GENERATED_H
+#define RUNTIME_COMPILER_GENERATED_H
+
+#define THROW_NOT_IMPLEMENTED throw std::runtime_error("Not implemented for tests");
```
Comment (projedi) [inline runtime/src/test_support/cpp/CompilerGenerated.h:9]: Maybe we can do without this macros? We can define a plain function `void ThrowNotImplementedForTests() { … }` here. However, I don't see anything wrong with repeating `throw std::runtime_error(...)` at each site either.

File: .gitignore [hunk 2]
```diff
@@ -62,6 +62,7 @@ llvmCoverageMappingC/CMakeLists.txt
 tools/performance-server/node_modules
 tools/performance-server/server
 tools/performance-server/ui/js
+runtime/cmake-build-debug/
```
Comment (projedi) [inline .gitignore:65]: Why is this needed?

File: runtime/src/test_support/cpp/CompilerGenerated.h (hunk 1 above)
Comment (ilmat192) [reply runtime/src/test_support/cpp/CompilerGenerated.h]: Fixed by inlining the macro.

File: .gitignore (hunk 2 above)
Comment (ilmat192) [reply .gitignore:65]: CMake generates this directory during CLion import

**Review by projedi — APPROVED** [review]
